### PR TITLE
Docstrings for some functions in classification metrics

### LINKED ISSUE #389: Add docstrings to protected functions
## 📚 Documentation

Add docstrings to all protected functions at least tell what are the arguments for...
This can see easily implemented in multiple PRs :rabbit:

### PR DESCRIPTION
**Pull request by DevPranjal**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?

Partially fixes #389 .

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Yes :] :)
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by DevPranjal** [comment]
Failing doctests, marking this draft for now. Thanks for your patience.

File: torchmetrics/functional/classification/hinge.py [hunk 1]
```diff
@@ -35,6 +35,17 @@ def _check_shape_and_type_consistency_hinge(
     preds: Tensor,
     target: Tensor,
 ) -> DataType:
+    """Checks shape and type of `preds` and `target` and returns mode of the input tensors. Raises `ValueError` if:
```
Comment (Borda) [inline torchmetrics/functional/classification/hinge.py:38]: Raises is separate section

File: torchmetrics/functional/classification/precision_recall.py [hunk 2]
```diff
@@ -27,6 +27,71 @@ def _precision_compute(
...
+
+    Example:
+        >>> from torchmetrics.functional.classification.stat_scores import _stat_scores_update
+        >>> preds  = torch.tensor([2, 0, 2, 1])
+        >>> target = torch.tensor([1, 1, 2, 0])
+        >>> tp, fp, tn, fn = _stat_scores_update(
+        ...                         preds,
+        ...                         target,
+        ...                         reduce='macro',
+        ...                         num_classes=3,
```
Comment (Borda) [inline torchmetrics/functional/classification/precision_recall.py:82]: could be a single line?

File: torchmetrics/functional/classification/accuracy.py [hunk 3]
```diff
@@ -78,6 +169,53 @@ def _accuracy_compute(
...
     mode: DataType,
 ) -> Tensor:
+    """Computes accuracy from stat scores: true positives, false positives, true negatives, false negatives.
+
+    Args:
+        tp: True positives
+        fp: False positives
+        tn: True negatives
+        fn: False negatives
+        average:
```
[the remark this answers is not in the document]
Comment (DevPranjal) [reply torchmetrics/functional/classification/accuracy.py:179]: Yes, I was wanting the same. Will change

File: torchmetrics/functional/classification/hinge.py (hunk 1 above)
Comment (DevPranjal) [reply torchmetrics/functional/classification/hinge.py:38]: Will add a separate section.

File: torchmetrics/functional/classification/precision_recall.py (hunk 2 above)
Comment (DevPranjal) [reply torchmetrics/functional/classification/precision_recall.py:82]: Yes

File: torchmetrics/functional/classification/auc.py [hunk 4]
```diff
@@ -18,6 +18,14 @@
 
 
 def _auc_update(x: Tensor, y: Tensor) -> Tuple[Tensor, Tensor]:
+    """Updates and returns variables required to compute area under the curve. Checks if the 2 input tenseorhave
+    the same number of elements and if they are 1d.
```
Comment (SkafteNicki) [inline torchmetrics/functional/classification/auc.py:22]:
```suggestion
    """Updates and returns variables required to compute area under the curve. Checks if the 2 input tensors 
    have the same number of elements and if they are 1d.
```

File: torchmetrics/functional/classification/auc.py [hunk 5]
```diff
@@ -36,12 +44,44 @@ def _auc_update(x: Tensor, y: Tensor) -> Tuple[Tensor, Tensor]:
...
+    Args:
+        x: x-coordinates, must be either increasing or decreasing, else a ValueError is raised
+        y: y-coordinates
+        direction: 1 if increaing, -1 if decreasing
+
+    Example:
+        >>> x = torch.tensor([0, 1, 2, 3])
+        >>> y = torch.tensor([0, 1, 2, 2])
+        >>> x, y = _auc_update(x, y)
+        >>> _auc_compute_without_check(x, y)
```
Comment (SkafteNicki) [inline torchmetrics/functional/classification/auc.py:58]: missing parameter
```suggestion
        >>> _auc_compute_without_check(x, y, 1.0)
```

File: torchmetrics/functional/classification/f_beta.py [hunk 6]
```diff
@@ -37,6 +37,70 @@ def _fbeta_compute(
     average: str,
     mdmc_average: Optional[str],
 ) -> Tensor:
+    """Computes f_beta metric from stat socres: true positives, false positives, true negatives, false negatives.
```
Comment (SkafteNicki) [inline torchmetrics/functional/classification/f_beta.py:40]:
```suggestion
    """Computes f_beta metric from stat scores: true positives, false positives, true negatives, false negatives.
```

**Review by Borda — APPROVED** [review]
NICE!, thank you

**Review by SkafteNicki — APPROVED** [review]

**Comment by SkafteNicki** [comment]
@DevPranjal thanks for sending these PRs :]
